### ISSUE TITLE
ZeroDivisionError - fastp module breaks when FastQ has 0 raw reads

### ISSUE THREAD

**Issue by drpatelh** [ewels/MultiQC#1444]
I tested with MultiQC `v1.10`. Creating a fresh folder with just the file below and running `multiqc .` should reproduce the error below:

```console
[WARNING]         multiqc : MultiQC Version v1.10.1 now available!
[INFO   ]         multiqc : This is MultiQC v1.10
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching   : viralrecon/testing
Searching   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 1/1
[ERROR  ]         multiqc : Oops! The 'fastp' MultiQC module broke...
  Please copy the following traceback and report it at https://github.com/ewels/MultiQC/issues
  If possible, please include a log file that triggers the error - the last file found was:
    ./test1.fastp.json
============================================================
Module fastp raised an exception: Traceback (most recent call last):
  File "conda/envs/multiqc-1.10/lib/python3.9/site-packages/multiqc/multiqc.py", line 579, in run
    output = mod()
  File "conda/envs/multiqc-1.10/lib/python3.9/site-packages/multiqc/modules/fastp/fastp.py", line 47, in __init__
    self.parse_fastp_log(f)
  File "conda/envs/multiqc-1.10/lib/python3.9/site-packages/multiqc/modules/fastp/fastp.py", line 201, in parse_fastp_log
    self.fastp_data[s_name]["after_filtering_total_reads"]
ZeroDivisionError: float division by zero
============================================================
[WARNING]         multiqc : No analysis results found. Cleaning up..
[INFO   ]         multiqc : MultiQC complete

```

Added `.txt` extension to be able to upload here.

[test1.fastp.json.txt](https://github.com/ewels/MultiQC/files/6625100/test1.fastp.json.txt)

**Comment by ewels** [comment]
_@drpatelh, pushing the limits of every MultiQC module by running with zero reads_ 😆 

Feel free to take a stab at it - these are usually fairly easy to fix with a `try` / `except ZeroDivisionError`... Otherwise will get to it when I can 👍🏻

**Comment by drpatelh** [comment]
Fix one and find another one. Sorry! No rush.

**Comment by ewels** [comment]
Fixed by @ErikDanielsson and me in #1451 👍🏻
